box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java

```diff
@@ -530,18 +541,62 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
                 public void run() {
                     //TODO decide whether to use two views for crossfading animation.
                     imageView.setImageBitmap(bitmap);
+                    ImageLoadListener listener = ViewData.getImageLoadListener(imageView);
+                    if(listener != null) {
+                        listener.onSuccess();
+                    }
                 }
             });
         }
 
     }
 
-
-
     private boolean isRequestStillApplicable(final BoxRequest request, final ImageView view) {
         return view.getDrawable() instanceof LoaderDrawable &&
                ((LoaderDrawable) view.getDrawable()).matchesRequest(request);
 
     }
 
+    /**
+     * Interface to listen for Image load, so that the calling component can
+     *  act upon error or success
+     */
+    public interface ImageLoadListener {
+        void onSuccess();
+        void onError();
+    }
+
+    /**
+     * Internal class to keep ImageView data information, like type of thumbnail to show, and image load listener
+     */
+    private static class ViewData {
+        private ImageLoadListener mListener;
+        private String mImageType;
+
+        ViewData(String type, ImageLoadListener listener) {
+            mImageType = type;
+            mListener = listener;
+        }
+
+        static ImageLoadListener getImageLoadListener(ImageView view) {
```

Comment:
Is there any advantage to have those 2 static function here (getImageLoadListener and getImageType)? Why not just put them (not static) in ThumbnailManager.java?

Reply:
As the data is owned by ViewData class, I thought it make more sense to have getters in there.